src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -507,5 +507,26 @@ public static int countEmptyStrings(String... strings) {
      */
     public static String convertToEmptyStringIfNull(String str) {
         return (str == null) ? "" : str;
-    }  
+    }
+    
+    /**
+     * Removes the outermost enclosing square brackets surrounding a string.
+     * 
+     * @param str
+     * @return the string without the outermost enclosing square brackets and 
+     *         without leading and trailing spaces, if the given string is enclosed by square brackets <br/> 
+     *         the string itself if the given string is not enclosed by square brackets <br/>
+     *         null if the given string is null
+     */
```

Comment:
Even pre-trimming is not necessary. There is no situation where the caller thinks the parameter is bracket enclosed but there are trailing/leading spaces that got there "accidentally" (such things can happen only in cases where the parameter is typed by a user). Remove the pre-trimming and get rid of unnecessary comments/test cases that are related to trimming.